var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -163,13 +163,16 @@ class Hpx(CMakePackage, CudaPackage, ROCmPackage):
     patch('git_external.patch', when='@1.3.0 instrumentation=apex')
 
     def instrumentation_args(self):
-        for value in self.variants['instrumentation'].values:
+        args = []
+        for value in self.variants['instrumentation'][0].values:
```

Comment:
is this really the best api we can provide :sweat_smile: it looks rather awkward

Reply:
Agreed -- maybe we should switch to NamedTuple for the variants so we can refer to them as `variant` and `conditions`? This is my fault on the initial PR.

Reply:
Well, we shouldn't be pulling from packages directly. Usually what we do is `spec.variants['name'].value` There are just a handful of packages that are reading the attribute from the package and not from the spec.